orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableTask.java

```diff
@@ -55,6 +57,20 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(30);
   }
 
+  @Override
+  public long getDynamicTimeout(StageExecution stage) {
+    ManifestTimeout timeout = stage.mapTo(ManifestTimeout.class);
+    if (timeout.timeoutInMinutes != null) {
+      return TimeUnit.MINUTES.toMillis(timeout.timeoutInMinutes.get());
+    }
+
+    return getTimeout();
+  }
+
+  static class ManifestTimeout {
```

Comment:
This class should be `private` if it's not intended to be used outside of the outer class.